src/models/RouterRouteHooks.ts

```diff
@@ -0,0 +1,10 @@
+import { RouterAfterRouteHook, RouterBeforeRouteHook, Routes } from '@/main'
```

Comment:
this import surprised me a bit, I wonder if that's intentional? Kinda feels like using barrels again